.evergreen/check-rustdoc.sh

```diff
@@ -3,6 +3,16 @@
 set -o errexit
 
 . ~/.cargo/env
+
+# docs.rs builds the driver on a read-only file system. to create a more realistic environment, we first
+# build the driver to ensure we have all the deps already in src, and then limit the permissions on that directory.
+# this is to help us avoid introducing problems like those described here 
+# https://docs.rs/about/builds#read-only-directories where we or a dependency modify source code during the
+# build process.
+cargo +nightly build
+cargo clean
+chmod -R 555 ~/.cargo/registry/src
```

Comment:
so you don't have to look it up, this gives read and execute, but not write, permission, to everyone.